composer.json

```diff
@@ -67,8 +69,13 @@
         "lint": "parallel-lint src tests",
         "phpunit": "phpunit --verbose --colors=always",
         "phpcs": "phpcs src tests --standard=psr2 -sp --colors",
+        "phpstan": [
+            "@php vendor/phpstan/phpstan-shim/phpstan.phar analyse -c phpstan.neon src --level 2 --no-progress",
+            "@php vendor/phpstan/phpstan-shim/phpstan.phar analyse -c phpstan-tests.neon tests --level 2 --no-progress"
+        ],
         "test": [
             "@lint",
+            "@phpstan",
```

Comment:
it should be ran before tests, because it can discover issues that would make tests fail anyway